driver/generic/st77xx.py

```diff
@@ -0,0 +1,417 @@
+import time
+import machine
+import struct
+import uctypes
+
+from micropython import const
+
+#
+# This driver was written from scratch using datasheets and looking at other drivers listed here:
```

Comment:
When you base your code on code of others, please pay attention to the legal aspects:
- Make sure you copy only from sources with permissive license (such as MIT, BSD etc.) and not GPL or LGPL or others which are incompatible with LVGL permissive (MIT) license. 
- Make sure you include the original license when needed.

For example, Adafruit ST7789 license is MIT, but [requires to include the copyright notice](https://github.com/pimoroni/st7789-python/blob/3e63dc34042c0d12bf2950ed5b88d3a8dc060018/library/ST7789/__init__.py#L11-L12):
```
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
```

Reply:
Done in the last commit. The Xpt2046 is not "based on" the driver linked, it was just an inspiration (like most of the others, anyway).